components/sync/service/brave_sync_service_impl_unittest.cc

```diff
@@ -708,4 +712,42 @@ TEST_F(BraveSyncServiceImplTest, NoLeaveDetailsWhenInitializeIOS) {
   EXPECT_FALSE(brave_sync_prefs()->GetLeaveChainDetails().empty());
 }
 
+TEST_F(BraveSyncServiceImplTest, OnAccountsCookieDeletedByUserAction) {
+  SyncTransportDataPrefs::RegisterProfilePrefs(pref_service()->registry());
+
+  SyncTransportDataPrefs sync_transport_data_prefs(
+      pref_service(), signin::GaiaIdHash::FromGaiaId("user_gaia_id"));
+
+  sync_transport_data_prefs.SetCacheGuid(kCacheGuid);
+
+  CreateSyncService();
+
+  brave_sync_service_impl()->OnAccountsCookieDeletedByUserAction();
+
+  EXPECT_EQ(sync_transport_data_prefs.GetCacheGuid(), kCacheGuid);
+}
+
+TEST_F(BraveSyncServiceImplTest, OnAccountsInCookieUpdated) {
```

Comment:
Since these test scenario are basically the same except observer function call, we can use `INSTANTIATE_TEST_SUITE_P` with `testing::ValuesIn` which contains a enum that contains these scenarios we want to test and act accordingly with `GetParam()`

Reply:
thanks, redone at https://github.com/brave/brave-core/pull/25971/commits/56f9ed040b5c649ce446fd28a6c80454839278f5